RocketChat/Rocket.Chat.ReactNative
# Chore: Migrate views/RoomListView to typescript

## Proposed changes
Migrate methods/RoomListView to typescript


## Issue(s)	
N/A

## How to test or reproduce
On App: 
* Check the list of chats; 
* Check the search rooms;
* Enter some chat and check if:
  * Check if the headers still works fine;
* Change the theme app, and verify if the chats list still works fine;


## Screenshots
N/A

## Types of changes




## Checklist



## Further comments
N/A
